node/src/components/block_proposer/tests.rs

```diff
@@ -483,24 +469,24 @@ fn should_not_propose_deploy_if_block_size_limit_within_threshold() {
         block_gas_limit: 10,
         max_transfer_count: 3,
         max_deploy_count: 3,
-        proposed_count: 2,
-        remaining_pending_count: 2,
+        proposed_count: 4,
+        remaining_pending_count: 0,
         max_block_size: Some(2 * DEPLOY_APPROX_MIN_SIZE),
     });
 }
 
 #[test]
 fn should_not_propose_deploy_if_block_size_limit_passed() {
     test_proposer_with(TestArgs {
-        deploy_count: 0,
-        transfer_count: 1,
+        deploy_count: 3,
+        transfer_count: 2, // transfers should -not- count towards the block size limit
```

Comment:
Can you add a test that sets a block size limit lower than `max_transfer_count` occupies? i.e. show that transfers can exceed block size limit.